### PR TITLE
Define top level Permission for ci.yml workflow 

### PR DESCRIPTION
**Pull request by Akash-Singh04**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #5048

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ on:
 concurrency:
   group: ${{ github.workflow }}-${{ github.actor }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
+permissions:
+  contents: read  # Adjust permissions as necessary
+  id-token: write # For OIDC tokens
```
Comment (zhzhuang-zju) [inline .github/workflows/ci.yml:17]:
```suggestion
  contents: read  # for actions/checkout to fetch code
  id-token: write # for requesting the JWT
```

**Review by zhzhuang-zju — COMMENTED** [review]
thanks, others LGTM

**Comment by Akash-Singh04** [comment]
@zhzhuang-zju I have updated the commits to pass the tests, let me know if i should squash all of them

**Comment by zhzhuang-zju** [comment]
> @zhzhuang-zju I have updated the commits to pass the tests, let me know if i should squash all of them

Yes, squash all of them would be better

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,6 +12,9 @@ on:
 concurrency:
   group: ${{ github.workflow }}-${{ github.actor }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
+permissions:
+  contents: read  # for actions/checkout to fetch code  
+  id-token: write # for requesting the JWT
```
Comment (zhzhuang-zju) [inline .github/workflows/ci.yml:17]: refer to https://github.com/karmada-io/karmada/pull/5119#discussion_r1662065591, the write permissions can be declared at the [run-level](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idpermissions). Also, can you confirm whether write permission is required? The result given by the [StepSecurity's online tool](https://app.stepsecurity.io/secureworkflow/) is that write permission is not required.

File: README.md [hunk 3]
```diff
@@ -13,6 +13,7 @@
 [![codecov](https://codecov.io/gh/karmada-io/karmada/branch/master/graph/badge.svg?token=ROM8CMPXZ6)](https://codecov.io/gh/karmada-io/karmada)
 [![FOSSA Status](https://app.fossa.com/api/projects/custom%2B28176%2Fgithub.com%2Fkarmada-io%2Fkarmada.svg?type=shield)](https://app.fossa.com/projects/custom%2B28176%2Fgithub.com%2Fkarmada-io%2Fkarmada?ref=badge_shield)
 [![Artifact HUB](https://img.shields.io/endpoint?url=https://artifacthub.io/badge/repository/karmada)](https://artifacthub.io/packages/krew/krew-index/karmada)
+[![CLOMonitor](https://img.shields.io/endpoint?url=https://clomonitor.io/api/projects/cncf/karmada/badge)](https://clomonitor.io/projects/cncf/karmada)
```
Comment (zhzhuang-zju) [inline README.md:16]: https://github.com/karmada-io/karmada/pull/5054 has added CLOMonitor badge

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Akash-Singh04) [reply .github/workflows/ci.yml:17]: That would be correct, write permission is not necessary

**Comment by Akash-Singh04** [comment]
@zhzhuang-zju Your requested changes have been made

**Review by zhzhuang-zju — COMMENTED** [review]
@Akash-Singh04 please squash the commits. and then LGTM

**Comment by zhzhuang-zju** [comment]
kindly ping @Akash-Singh04

**Comment by RainbowMango** [comment]
I just helped rebase this PR, please @zhzhuang-zju take a look.

**Comment by zhzhuang-zju** [comment]
> I just helped rebase this PR, please @zhzhuang-zju take a look.

thanks~ 
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**